cam_diagnostics/cam_diagnostics.F90

```diff
@@ -0,0 +1,199 @@
+                                               'CLDICE', &
+                                               'RAINQM'/)
+
+CONTAINS
+
+   !> \section arg_table_cam_state_diagnostics_init  Argument Table
+   !! \htmlinclude cam_state_diagnostics_init.html
+   subroutine cam_state_diagnostics_init(const_props, errmsg, errflg)
+      use cam_history,         only: history_add_field
+      use cam_history_support, only: horiz_only
+
+      type(ccpp_constituent_prop_ptr_t), intent(in) :: const_props(:)
+      character(len=512), intent(out) :: errmsg
+      integer,            intent(out) :: errflg
+
+      ! Local variables:
+
+      integer :: const_idx, name_idx
+      character(len=512) :: standard_name
+
+      errmsg = ''
+      errflg = 0
+
+      ! Add state fields
+      call history_add_field('PS',        'surface_pressure',                           horiz_only,  'avg', 'Pa')
+      call history_add_field('PSDRY',     'surface_pressure_of_dry_air',                horiz_only,  'avg', 'Pa')
+      call history_add_field('PHIS',      'surface_geopotential',                       horiz_only,  'lst', 'Pa')
+      call history_add_field('T',         'air_temperature',                                 'lev',  'avg', 'K')
+      call history_add_field('U',         'eastward_wind',                                   'lev',  'avg', 'm s-1')
+      call history_add_field('V',         'northward_wind',                                  'lev',  'avg', 'm s-1')
+      call history_add_field('DSE',       'dry_static_energy',                               'lev',  'avg', 'm s-1')
+      call history_add_field('OMEGA',     'lagrangian_tendency_of_air_pressure',             'lev',  'avg', 'Pa s-1')
+      call history_add_field('PMID',      'air_pressure',                                    'lev',  'avg', 'Pa')
+      call history_add_field('PMIDDRY',   'air_pressure_of_dry_air',                         'lev',  'avg', 'Pa')
+      call history_add_field('PDEL',      'air_pressure_thickness',                          'lev',  'avg', 'Pa')
+      call history_add_field('PDELDRY',   'air_pressure_thickness_of_dry_air',               'lev',  'avg', 'Pa')
+      call history_add_field('RPDEL',     'reciprocal_of_air_pressure_thickness',            'lev',  'avg', 'Pa-1')
+      call history_add_field('RPDELDRY',  'reciprocal_of_air_pressure_thickness_of_dry_air', 'lev',  'avg', 'Pa-1')
+      call history_add_field('LNPMID',    'ln_air_pressure',                                 'lev',  'avg', '1')
+      call history_add_field('LNPMIDDRY', 'ln_air_pressure_of_dry_air',                      'lev',  'avg', '1')
+      call history_add_field('EXNER',     'inverse_exner_function_wrt_surface_pressure',     'lev',  'avg', '1')
+      call history_add_field('ZM',        'geopotential_height_wrt_surface',                 'lev',  'avg', 'm')
+      call history_add_field('PINT',      'air_pressure_at_interface',                       'ilev', 'avg', 'Pa')
```

Comment:
Just FYI that in the official CCPP standard names dictionary interface variables now end with `at_interfaces` (note the extra "s"), so it might be good to change the long names here even if it will be a little while until we update the standard names themselves.

Reply:
updated!